pkg/cache/redis_cache.go

```diff
@@ -0,0 +1,89 @@
+package cache
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/redis/go-redis/v9"
+)
+
+type (
+	RedisCache[O any] struct {
+		client     *redis.Client
+		expiration time.Duration
+		fetch      FetchFunc[string, O]
+		prefix     string
+	}
+)
+
+func NewRedis[O any](client *redis.Client, prefix string, expiration time.Duration, fetch FetchFunc[string, O]) *RedisCache[O] {
+	return &RedisCache[O]{
+		client:     client,
+		expiration: expiration,
+		fetch:      fetch,
+		prefix:     prefix,
+	}
+}
+
+func (c *RedisCache[O]) prefixedKey(key string) string {
+	return c.prefix + key
```

Comment:
prefix could be inherited from O in my opinion, this would prevent accidental clashes